.github/pull_request_template.md

```diff
@@ -0,0 +1,42 @@
+#### What does the PR do?
+<!-- Describe your pull request here. Please read the text below the line, and make sure you follow the checklist.-->
+
+
+#### Checklist
+- [ ] Populated all fields in this template.
+- [ ] The PR title reflects the change.
+- [ ] Changes are described in the pull request, or an [existing issue is referenced](https://github.com/triton-inference-server/server/issues).
+- [ ] Added [test plan](#test-plan) and verified test passes.
+- [ ] Verified that the PR passes existing CI.
+- [ ] Added _succinct_ git squash message before merging [ref](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+- [ ] CodeQL suggestions resolved
+- [ ] Lint Gods are pleased ( i.e. pre-commit checks are green )
+
+#### Type:
```

Comment:
For the "Types", should we try to make use of Github's label feature since both seem to serve similar purpose? Seems like we can also edit the label names to match with our needs.

Reply:
Pushing back on lables since they are also used by github issues and it could be confusing to other people what the labels will be used.

Reply:
If there is a standard way to see PRs for "features, Bugs, enhancement" by labels - I think it'd be worth it. Seems like context (issues vs PRs) would be clear in terms of what you are searching for. 

might get additional benefit from github filtering if we reuse labels. 

@nv-hwoo - do you have an example?

Reply:
@nnshah1 @jbkyang-nvi Yeah labels are used both in issues and PRs but they can be filtered by key values. For instance, for enhancement label, we can use `is:issue` and `label:enhancement` to search for issues with enhancement label. Similarly, we can use `is:pr` and `label:enhancement` to filter PRs with the label. 

Haven't used Github labels that much so I'm not sure what's the best practice is (if there's any) but just thought they could be useful 🙂